test/fuzz/vclock_from_string_fuzzer.c

```diff
@@ -0,0 +1,27 @@
+#include <stdlib.h>
+#include <stdint.h>
+#include <stddef.h>
+
+#include "vclock/vclock.h"
+
+int
+LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+	char *buf = xcalloc(size + 1, sizeof(char));
+	if (!buf)
+		return 0;
+	memcpy(buf, data, size);
+	buf[size] = '\0';
+
+	struct vclock vclock;
+	vclock_create(&vclock);
+	size_t res = vclock_from_string(&vclock, buf);
+	const char *vclock_str = vclock_to_string(&vclock);
+	if (size == res) {
```

Comment:
It seems `vclock_from_string` returns 0 when everything's good and amount of successfully parsed bytes otherwise:
https://github.com/tarantool/tarantool/blob/f7bb3fc7a1222107753dc420675ecda043b0a5d2/src/lib/vclock/vclock.c#L150-L161

Reply:
Also by looking at the code of `vclock_from_string` it seems it accepts vclocks like `{    1  : 2}` (with many spaces in between), but this input string won't match what is output from `vclock_to_string`.

Reply:
I think making sure input and output strings are the same would be quite a hustle.

You could check that `vclock_from_string(intput)` is the same as `vclock_from_string(vclock_to_string(input))` instead

Reply:
Updated version doesn't use returned value.